third_party/terraform/website/docs/r/sql_database_instance.html.markdown

```diff
@@ -100,17 +104,29 @@ resource "random_id" "db_name_suffix" {
 }
 
 resource "google_sql_database_instance" "postgres" {
-  name = "postgres-instance-${random_id.db_name_suffix.hex}"
+  name             = "postgres-instance-${random_id.db_name_suffix.hex}"
   database_version = "POSTGRES_9_6"
 
   settings {
     tier = "db-f1-micro"
 
     ip_configuration {
-      authorized_networks = [
-        "${data.null_data_source.auth_netw_postgres_allowed_1.*.outputs}",
-        "${data.null_data_source.auth_netw_postgres_allowed_2.*.outputs}",
-      ]
+      dynamic "authorized_networks" {
+        for_each = [data.null_data_source.auth_netw_postgres_allowed_1.*.outputs]
+        content {
+          expiration_time = lookup(authorized_networks.value, "expiration_time", null)
+          name            = lookup(authorized_networks.value, "name", null)
+          value           = lookup(authorized_networks.value, "value", null)
+        }
+      }
+      dynamic "authorized_networks" {
+        for_each = [data.null_data_source.auth_netw_postgres_allowed_2.*.outputs]
+        content {
+          expiration_time = lookup(authorized_networks.value, "expiration_time", null)
+          name            = lookup(authorized_networks.value, "name", null)
+          value           = lookup(authorized_networks.value, "value", null)
+        }
+      }
```

Comment:
`terraform 0.12upgrade` left this comment here:

>  TF-UPGRADE-TODO: The automatic upgrade tool can't predict
>  which keys might be set in maps assigned here, so it has
>  produced a comprehensive set here. Consider simplifying
>  this after confirming which keys can be set in practice.

After looking above do we not need `expiration_time` set in these?

Reply:
I'm not too familiar with this example, but from my reading the goal is to create a list of IP addresses including all of the compute_instance ip's and the [two manual ones](https://github.com/GoogleCloudPlatform/magic-modules/pull/2574/files#diff-18e82d50a24d79aee2e6bd023c4b3905R98). Then make sure that all of those ip's are listed as authorized networks and throws a `name` in there for good measure. So I feel pretty confident that `expiration_time` is not needed.
However I'm less confident about the the dynamic block syntax, but it seems like we could get away with removing the null resources here completely and accomplish the same goal. Which I think would be a great example of "here's something we had to work around in 0.11 that isn't such a hack in 0.12".